src/Domain/DomainHelpers.hpp

```diff
@@ -0,0 +1,329 @@
+
+  /*!
+   * Takes the Directions in block1 and finds the corresponding
+   * Directions in block2 according to their corner numberings.
+   */
+  static std::pair<std::array<Direction<VolumeDim>, VolumeDim>,
+                   std::array<Direction<VolumeDim>, VolumeDim>>
+  obtain_correspondence_between_blocks(
+      size_t block1_id, size_t block2_id,
+      const std::vector<std::array<size_t, two_to_the(VolumeDim)>>&
+          corners_of_all_blocks);
+
+  /*!
+   * Given Directions in block1 and their corresponding Directions
+   * in block2, this function repackages them into a std::pair.
+   */
+  static std::pair<std::array<Direction<VolumeDim>, VolumeDim>,
+                   std::array<Direction<VolumeDim>, VolumeDim>>
+  create_correspondence_between_blocks(
+      const Direction<VolumeDim>& direction_to_guest_in_host_frame,
+      const Direction<VolumeDim>& direction_to_host_in_guest_frame,
+      const std::vector<Direction<VolumeDim>>&
+          alignment_of_shared_face_in_host_frame,
+      const std::vector<Direction<VolumeDim>>&
+          alignment_of_shared_face_in_guest_frame);
+
+  /*!
+   *  Calculates the RMS separation between two blocks in a domain via their
+   *  EmbeddingMaps, to confirm they are physically connected.
+   */
+  static double physical_rms_separation(
+      const EmbeddingMap<VolumeDim, VolumeDim>& map_of_block1,
+      const EmbeddingMap<VolumeDim, VolumeDim>& map_of_block2,
+      const std::array<size_t, two_to_the(VolumeDim)>& corners_of_block1,
+      const std::array<size_t, two_to_the(VolumeDim)>& corners_of_block2);
+
+  /*!
+   *  When setting up periodic-like boundary conditions, the user specifies
+   *  the corners of the two faces to be identified.
+   *  \returns the corresponding id of the block whose faces are to
+   *  be identified.
+   *  \requires that the two faces correspond to the same block.
+   */
+  static size_t find_block_id(
+      const std::array<size_t, two_to_the(VolumeDim - 1)>& global_corners_face1,
+      const std::array<size_t, two_to_the(VolumeDim - 1)>& global_corners_face2,
+      const std::vector<std::array<size_t, two_to_the(VolumeDim)>>&
+          corners_of_all_blocks);
+
+  /*!
+   *  Sets up the BlockNeighbors using the corner numbering scheme
+   *  to deduce the correct neighbors and orientations. Does not set
+   *  up periodic-like boundary conditions.
+   */
+  static void set_internal_boundaries(
+      const std::vector<std::unique_ptr<EmbeddingMap<VolumeDim, VolumeDim>>>&
+          maps_of_all_blocks,
+      const std::vector<std::array<size_t, two_to_the(VolumeDim)>>&
+          corners_of_all_blocks,
+      std::vector<
+          std::unordered_map<Direction<VolumeDim>, BlockNeighbor<VolumeDim>>>&
+          neighbors_of_all_blocks);
+
+  /*!
+   *  Sets up additional BlockNeighbors corresponding to any
+   *  periodic-like boundary condtions provided by the user.
+   */
+  static void set_periodic_like_boundaries(
```

Comment:
periodic_like for mobius strip / klein bottle like boundary conditions (but periodic_like is only my own name for those BCs)